bns-test/test/name_register.test.ts

```diff
@@ -0,0 +1,345 @@
+          expect(receipt.success).eq(false);
+      //   });
+
+      // });
+
+      // describe("Charlie registering 'bob.blockstack'", () => {
+
+      //   it("should fail", async () => {
+
+          receipt = await bns.namePreorder(
+            cases[0].namespace,
+            "bob",
+            cases[0].salt, 
+            2560000, { sender: charlie });
+          expect(receipt.success).eq(true);
+          expect(receipt.result).include('u50');
+  
+          receipt = await bns.nameRegister(
+            cases[0].namespace, 
+            "bob", 
+            cases[0].salt, 
+            cases[0].zonefile, { sender: charlie });
+          expect(receipt.error).include('2004');
+          expect(receipt.success).eq(false);
+      //   });
+      // });
+
+      // // todo(ludo): investigate the case where 'name.namespace' was preordered while being unavailable
+      // // and then became available
+      // describe("Bob registering a second name 'bobby.blockstack'", () => {
+
+        // it("should fail if 'bob.blockstack' is not expired", async () => {
+          await bns.mineBlocks(1);
+
+          receipt = await bns.namePreorder(
+            cases[0].namespace,
+            "bobby",
+            cases[0].salt, 
+            160000, { sender: cases[0].nameOwner });
+          expect(receipt.success).eq(true);
+          expect(receipt.result).include('u53');
+  
+          receipt = await bns.nameRegister(
+            cases[0].namespace, 
+            "bobby", 
+            cases[0].salt, 
+            cases[0].zonefile, { sender: cases[0].nameOwner });
+          expect(receipt.error).include('3001');
+          expect(receipt.success).eq(false);
+        // });
+
+        // it("should succeed once 'bob.blockstack' is expired", async () => {
+          await mineBlocks(bns, cases[0].renewalRule);
+
+          receipt = await bns.namePreorder(
+            cases[0].namespace,
+            "bobby",
+            cases[0].salt, 
+            160000, { sender: cases[0].nameOwner });
+          expect(receipt.success).eq(true);
+          expect(receipt.result).include('u65');
+  
+          receipt = await bns.getNameZonefile(
+            cases[0].namespace, 
+            "bob", { sender: cases[0].nameOwner });
+
+          receipt = await bns.nameRegister(
+            cases[0].namespace, 
+            "bobby", 
+            cases[0].salt, 
+            cases[0].zonefile, { sender: cases[0].nameOwner });
+          expect(receipt.result).include('true');
+          expect(receipt.success).eq(true);
+
+          receipt = await bns.getNameZonefile(
+            cases[0].namespace, 
+            "bobby", { sender: cases[0].nameOwner });
+          expect(receipt.result).include('0x30303030');
+          expect(receipt.success).eq(true);
+      //   });
+      // });
+
+      // describe("Charlie registering 'bob.blockstack'", () => {
+
+      //   it("should succeed once 'bob.blockstack' is expired", async () => {
```

Comment:
There should be a renewal grace period after `bob.blockstack` stops resolving, right?

Reply:
Yeah you'll find some test vectors on the renewal during grace period in `name-renewal.test.ts`